activation/nipst.go

```diff
@@ -148,6 +148,15 @@ func NewNIPSTBuilder(
 	}
 }
 
+// UpdatePoETProver updates poetProver reference. It should not be executed concurently with BuildNIPST.
```

Comment:
how do we enforce this rule?

Reply:
i don't think that this object should be public at all. it stores some state already so even now if someone will try to BuildNiPST from 2 goroutines the behavior is not safe

Reply:
actually i will make this method private for this module